Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp

```diff
@@ -6476,8 +6476,14 @@ OwnPtr<CalculationNode> Parser::parse_a_calculation(Vector<ComponentValue> const
             // FIXME: Resolutions, once calc() supports them.
             else if (dimension->is_time())
                 values.append({ NumericCalculationNode::create(dimension->time()) });
-            else
+            else if (dimension->is_flex()) {
+                // https://www.w3.org/TR/css3-grid-layout/#fr-unit
+                // NOTE: <flex> values are not <length>s (nor are they compatible with <length>s, like some <percentage> values),
+                //       so they cannot be represented in or combined with other unit types in calc() expressions.
+                return nullptr;
```

Comment:
I believe this part of that spec is outdated. Not combining with other unit types is fine, that's standard calc behaviour, but flex values should be allowed in calc. For example the values-4 spec mentions them in the algorithm for determining the type of a calculation: https://www.w3.org/TR/css-values-4/#calc-type-checking

(Yay for specs being inconsistent.)

So if I'm not mistaken, this should return a node wrapping the flex value. (My bad for missing that before. 😅)

Reply:
I checked both chrome and firefox and they both treat `grid-template-columns: calc(1fr);` as "Invalid property value". I don't think we want to diverge from other popular engines.

Reply:
That sounds more like they just don't support it yet.

Reply:
yes, that's probably true. although, for now, it should be ok to do what other engines do which is better than crashing 😅 also adding support for this will make us produce different grid layout compared to other engines and I would prefer to avoid that.

Reply:
I think both of you are right, but let's get this merged to avoid crashing, and we can look at supporting `fr` units in `calc()` separately :catdog: